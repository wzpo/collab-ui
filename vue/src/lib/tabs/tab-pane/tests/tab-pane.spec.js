import { mount } from '@vue/test-utils';
import TabPane from '../index.vue';

describe('TabPane', () => {
  it('should match snapshot', () => {
    const container = mount(TabPane);
    expect(container.html()).toMatchSnapshot();
  });

  it('should render one TabPane', () => {
    const container = mount(TabPane);

    expect(container.findAll('.md-tab__pane').length).toEqual(1);
  });

  it('should render inactive by default', () => {
    const container = mount(TabPane);

    expect(container.find('.md-tab__pane').classes('active')).toEqual(false);
  });

  it('should render active when passed', () => {
    const container = mount(TabPane, {
      propsData: {
        active: true
      }
    });

    expect(container.find('.md-tab__pane').classes('active')).toEqual(true);
  });

  it('should render children', () => {
    const container = mount(TabPane, {
      slots: {
        default: `<div class="testingforTP" />`
      }
    });

    expect(container.findAll('.testingforTP').length).toEqual(1);
  });
});

