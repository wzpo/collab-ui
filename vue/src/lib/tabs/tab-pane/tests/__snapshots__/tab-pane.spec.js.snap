// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPane should match snapshot 1`] = `
<div class="md-tab__pane">
  <div class="md-tab__content"></div>
</div>
`;
